src/pages/today/index.tsx

```diff
@@ -25,32 +58,63 @@ export type HabitTodo = {
 export default function Today() {
 	const [date, setDate] = useState(new Date());
 	const [habitTodos, setHabitTodos] = useState<HabitTodo[]>([]);
+	const documentId = useRef<string>(null);
 
 	const completedHabits = habitTodos.filter((habit) => habit.isComplete);
 	const incompleteHabits = habitTodos.filter((habit) => !habit.isComplete);
 
 	const isDateToday = isToday(date);
 	const isDateYesterday = isYesterday(date);
 
+	const handleCheckHabit =
+		(habitId: string) => (e: React.ChangeEvent<HTMLInputElement>) =>
+			markHabitComplete(e.target.checked, habitId, documentId.current);
+
 	useEffect(() => {
-		const unsub = onSnapshot(
-			getDateDoc(date),
-			(doc) => {
-				if (doc.exists()) {
-					const { date, ...habits } = doc.data();
-					setHabitTodos(
-						Object.entries(habits).map(([id, habit]) => ({
-							id,
-							...(habit as Omit<HabitTodo, 'id'>),
-						}))
+		return onSnapshot(
+			query(
+				datesCollection(),
+				where('date', '>=', startOfDay(date)),
+				where('date', '<=', endOfDay(date)),
```

Comment:
any reason you can't do `where('date', '==', date)`?

Reply:
I tried that but it doesn't work. It seems like in firebase, the '==' doesn't work for dates.